src/compile/babel-compiler.js

```diff
@@ -0,0 +1,48 @@
+var babel = require('babel-core');
+var path = require('path');
+var Promise = require('bluebird');
+var sourceMap = require('../source-map-utility');
+
+/**
+ * @param {string[]} presets
+ * @param {object} options
+ * @param {string} options.code
+ * @param {string} options.inputPath
+ * @param {string} options.nodeModulesPath
+ * @param {boolean} options.sourceMap
+ * @param {string} options.siteRoot
+ * @returns {bluebird}
+ */
+function transform(presets, options) {

```

Comment:
This is a shared compilation task that both `compile-jsx` and `compile-es6` call. They just each pass in the set of babel presets to use.